feat(frontend): implement my reading page UI with question answers display

ericsen-tsai opened the pull request:
## Summary

This PR implements the "My Reading" page UI for the membership section, allowing users to view their question answers organized by article. The implementation includes a new accordion-based interface for displaying choice and essay answers, along with supporting components for authentication state management and navigation.

## Changes

- **New Membership Pages**: Added `/member` and `/myreading` routes with corresponding page components
- **Accordion Component**: Created reusable accordion component using `@radix-ui/react-accordion` for displaying grouped question answers
- **Post Question Answers List**: Implemented components to display user's choice and essay answers with proper formatting, including:
  - Choice answer items with correct/incorrect indicators
  - Essay answer items with like counts
  - Question grouping by article with date and answer count badges
- **Membership Side Menu**: Added navigation sidebar for membership section with active state highlighting
- **Header Authentication Integration**: 
  - Added `AuthHeaderLoggedInSetter` component to sync authentication state with header
  - Updated header to show member page link when logged in instead of login link
  - Implemented mobile back button functionality via `HeaderMobileBackButtonHrefSetter`
- **New Icons**: Added multiple icons including `ArrowDown`, `DefaultAvatarIcon`, `CorrectIcon`, `IncorrectIcon`, `ThumbUpIcon`, `LightbulbIcon`, `ArticleShortcutIcon`, and `GroupIcon`
- **Asset Updates**: Replaced `avatar_default.png` with `avatar_default.svg` for better scalability
- **Routing UI Package Updates**: 
  - Bumped version from `0.1.0-alpha.4` to `0.1.0-alpha.5`
  - Added `isLoggedIn` state management to header context
  - Added mobile back button href management
- **Dependencies**: Added `@radix-ui/react-accordion` package for accordion functionality
- **Layout Updates**: Added membership-specific layout and integrated `AuthHeaderLoggedInSetter` across multiple layouts

## Additional Notes

- The implementation uses mock data for development and testing purposes
- Pagination is implemented with 5 items per page
- The accordion supports multiple open items simultaneously
- Empty states are handled for both the main list and individual answer groups
- Responsive design considerations are included for mobile, tablet, and desktop views

## Screenshot(s)

<img width="1243" height="879" alt="image" src="https://github.com/user-attachments/assets/6101946c-e403-45c9-b57a-e9421ea2649d" />

https://github.com/user-attachments/assets/1a3acc41-0f47-4bd8-86d2-0711a38ae51d

## Reference(s)


- [Notion Page](https://www.notion.so/twreporter/UI-2a28dbdca932809c83ddf81477c71f4e?v=1588dbdca93281dfa43c000c8415265e&source=copy_link)
- [Figma Design](https://www.figma.com/design/FqWHr59xE8l9teNoCmJGVn/%E6%9C%83%E5%93%A1%E9%A0%81?node-id=482-145077&m=dev)

ericsen-tsai committed:
feat(routing-ui): add isLoggedIn and mobileBackButtonHref to header context

ericsen-tsai committed:
chore(routing-ui,frontend): bump routing-ui version to 0.1.0-alpha.5

ericsen-tsai committed:
feat(frontend): add auth header logged in setter

ericsen-tsai committed:
feat(frontend): replace default avatar PNG with SVG and implement membership layout

- Removed the old default avatar PNG and added a new SVG version.
- Created a new layout component for the membership section, including a header and authentication setter.
- Added a member page that utilizes the new layout and displays user information.
- Introduced a side menu for membership navigation and constants for menu items and mock user data.

ericsen-tsai committed:
feat(routing-ui): hide mobile back button on mobile

ericsen-tsai committed:
feat(frontend): add my reading skeleton UI

ericsen-tsai committed:
feat(frontend): integrate Radix UI accordion component

- Added Radix UI accordion component with Accordion, AccordionItem, AccordionTrigger, and AccordionContent.
- Updated package.json to include @radix-ui/react-accordion dependency.
- Introduced new icons for accordion functionality, including ArrowDown.
- Updated yarn.lock to reflect new dependencies.

ericsen-tsai committed:
feat(frontend): add membership types and mock data for post question answers

- Introduced a new TypeScript file defining the PostQuestionAnswers type.
- Added mock data for post question answers, including various posts with choice and essay answers, to support the membership module.

ericsen-tsai committed:
feat(frontend): add my reading feature with pagination and answer display

- Implemented pagination for post question answers in the My Reading section.
- Added new components for displaying choice and essay answers.
- Integrated an accordion layout for organizing question answers.

ericsen-tsai committed:
refactor(frontend): adjust my reading page UI styling

ericsen-tsai committed:
feat(frontend): make swei marker font local to prevent font bouncing

ericsen-tsai committed:
fix(frontend): fix answer selected state in QA modal

ericsen-tsai committed:
feat(frontend,cms): add loginUrl to header and auth header logged in setter

ericsen-tsai committed:
feat(frontend): remove deprecated login page and login test route

ericsen-tsai committed:
fix(frontend): fix copilot comments

- Included tw-animate-css package in package.json and yarn.lock.
- Updated globals.css to import tw-animate-css for enhanced styling.
- Refined accordion component styles for improved focus visibility and animation transitions.
- Adjusted various components to ensure consistent styling and functionality.
- Updated arrow icon color to current color.

ericsen-tsai commented on packages/frontend/src/modules/membership/my-reading/post-question-answers-list/index.tsx:
```diff
@@ -0,0 +1,94 @@
+      {postQuestionAnswers.map((group) => {
+        return (
+          <AccordionItem
+            key={group.title}
+            value={group.title}
```
Since post title is unique, I'll just leave it as it was.

ericsen-tsai commented on packages/frontend/src/components/accordion.tsx:
```diff
@@ -0,0 +1,66 @@
+      <AccordionPrimitive.Trigger
+        data-slot="accordion-trigger"
+        className={cn(
+          'focus-visible:border-ring focus-visible:ring-ring/50 flex flex-1 items-start justify-between gap-4 rounded-md py-4 text-left font-medium transition-all outline-none focus-visible:ring-[3px] disabled:pointer-events-none disabled:opacity-50 [&[data-state=open]>svg]:rotate-180',
```
Fixed in [fix(frontend): fix copilot comments](https://github.com/kids-reporter/kids-reporter-monorepo/pull/825/commits/94f3accb0bb40d93e94265c8a4e06df96949766a)

ericsen-tsai commented on packages/frontend/src/modules/membership/member/index.tsx:
```diff
@@ -0,0 +1,51 @@
+  const router = useRouter()
+
+  useEffect(() => {
+    if (!isMobile) {
```
Fixed in [fix(frontend): fix copilot comments](https://github.com/kids-reporter/kids-reporter-monorepo/pull/825/commits/94f3accb0bb40d93e94265c8a4e06df96949766a)

ericsen-tsai commented on packages/routing-ui/src/header/desktop-header.tsx:
```diff
@@ -98,7 +102,7 @@ export function DesktopHeader({
                 />
                 <a
-                  href="/login"
+                  href={isLoggedIn ? '/member' : loginUrl}
```
Fixed in [fix(frontend): fix copilot comments](https://github.com/kids-reporter/kids-reporter-monorepo/pull/825/commits/94f3accb0bb40d93e94265c8a4e06df96949766a)

ericsen-tsai commented on packages/frontend/src/modules/membership/my-reading/post-question-answers-list/question-answers-list-content.tsx:
```diff
@@ -0,0 +1,74 @@
+        const isChoiceAnswer = 'choiceIndex' in answer
+        return (
+          <div
+            key={index}
```
Fixed in [fix(frontend): fix copilot comments](https://github.com/kids-reporter/kids-reporter-monorepo/pull/825/commits/94f3accb0bb40d93e94265c8a4e06df96949766a)

ericsen-tsai commented on packages/frontend/src/modules/membership/my-reading/post-question-answers-list/choice-answer-item.tsx:
```diff
@@ -0,0 +1,55 @@
+  const correctOptionIndex = options.findIndex((opt) => opt.isCorrectAnswer)
+
+  return (
+    <div key={answer.id} className="flex flex-col">
```
Fixed in [fix(frontend): fix copilot comments](https://github.com/kids-reporter/kids-reporter-monorepo/pull/825/commits/94f3accb0bb40d93e94265c8a4e06df96949766a)

ericsen-tsai commented on packages/frontend/src/modules/membership/my-reading/post-question-answers-list/choice-answer-item.tsx:
```diff
@@ -0,0 +1,55 @@
+        </div>
+        <div className="flex flex-1 items-center rounded-r-xl bg-white px-4 py-4">
+          <p className="prose-p1-medium text-neutral-900">
+            {`${(answer.choiceIndex ?? 0) + 1}. ${selectedOption?.content}`}
```
Fixed in [fix(frontend): fix copilot comments](https://github.com/kids-reporter/kids-reporter-monorepo/pull/825/commits/94f3accb0bb40d93e94265c8a4e06df96949766a)

ericsen-tsai commented on packages/routing-ui/src/header/mobile-header.tsx:
```diff
@@ -28,12 +34,22 @@ export function MobileHeader({
             {!showCloseButton && (
               <a
-                href="/login"
+                href={isLoggedIn ? '/member' : loginUrl}
```
Fixed in [fix(frontend): fix copilot comments](https://github.com/kids-reporter/kids-reporter-monorepo/pull/825/commits/94f3accb0bb40d93e94265c8a4e06df96949766a)

ericsen-tsai commented on packages/frontend/src/modules/membership/my-reading/post-question-answers-list/essay-answer-item.tsx:
```diff
@@ -0,0 +1,21 @@
+function EssayAnswerItem({ answer }: { answer: PostEssayAnswer }) {
+  const likesCount = (answer.likesCount ?? 0) > 99 ? '99+' : answer.likesCount
+  return (
+    <div key={answer.id} className="flex gap-4 rounded-xl bg-white px-5 py-4">
```
Fixed in [fix(frontend): fix copilot comments](https://github.com/kids-reporter/kids-reporter-monorepo/pull/825/commits/94f3accb0bb40d93e94265c8a4e06df96949766a)

ericsen-tsai commented on packages/routing-ui/src/icons/index.tsx:
```diff
@@ -356,3 +356,23 @@ export const HamburgerIconSmall = () => (
+    >
+      <path
+        d="M8.25 15.25L1.25 8.25L8.25 1.25"
+        stroke="#8E8E8E"
```
Fixed in [fix(frontend): fix copilot comments](https://github.com/kids-reporter/kids-reporter-monorepo/pull/825/commits/94f3accb0bb40d93e94265c8a4e06df96949766a)

ericsen-tsai commented on packages/frontend/src/components/accordion.tsx:
```diff
@@ -0,0 +1,66 @@
+  children,
+  ...props
+}: React.ComponentProps<typeof AccordionPrimitive.Content>) {
+  return (
+    <AccordionPrimitive.Content
+      data-slot="accordion-content"
+      className="data-[state=closed]:animate-accordion-up data-[state=open]:animate-accordion-down overflow-hidden"
+      {...props}
+    >
+      <div className={cn('pt-0 pb-4', className)}>{children}</div>
```
I added correct animation library to handle it.
Fixed in [fix(frontend): fix copilot comments](https://github.com/kids-reporter/kids-reporter-monorepo/pull/825/commits/94f3accb0bb40d93e94265c8a4e06df96949766a)

ericsen-tsai commented on packages/routing-ui/src/header/header-context.tsx:
```diff
@@ -39,8 +50,27 @@ export function HeaderProvider({
+      mobileBackButtonHref,
+      setMobileBackButtonHref,
+      loginUrl,
+      setLoginUrl,
```
Fixed in [fix(frontend): fix copilot comments](https://github.com/kids-reporter/kids-reporter-monorepo/pull/825/commits/94f3accb0bb40d93e94265c8a4e06df96949766a)

ericsen-tsai committed:
fix(routing-ui): clear login url when component unmounts

ericsen-tsai commented on packages/routing-ui/src/header/is-logged-in-setter.tsx:
```diff
@@ -0,0 +1,24 @@
+
+function IsLoggedInSetter({ isLoggedIn, loginUrl }: IsLoggedInSetterProps) {
+  const context = useHeaderContext()
+  const setIsLoggedIn = context?.setIsLoggedIn
+  const setLoginUrl = context?.setLoginUrl
+
+  useEffect(() => {
+    setIsLoggedIn?.(isLoggedIn)
+    setLoginUrl?.(loginUrl)
+    return () => setIsLoggedIn?.(false)
+  }, [isLoggedIn, setIsLoggedIn, loginUrl, setLoginUrl])
```
Fixed in [fix(routing-ui): clear login url when component unmounts](https://github.com/kids-reporter/kids-reporter-monorepo/pull/825/commits/497db6c5f312c57703926d9ea5a22a8a5cf0cfb6)

ericsen-tsai commented on packages/frontend/src/icons/arrow.tsx:
```diff
@@ -1,57 +1,67 @@
+      <g clipPath="url(#clip0_535_16631)">
+        <path
+          d="M5 9L12 16L19 9"
+          stroke="#8E8E8E"
```
Fixed in [fix(frontend): fix copilot comments](https://github.com/kids-reporter/kids-reporter-monorepo/pull/825/commits/94f3accb0bb40d93e94265c8a4e06df96949766a)

ericsen-tsai committed:
fix(routing-ui): remove unused swei font family

nickhsine commented on packages/frontend/src/app/(membership)/layout.tsx:
```diff
@@ -0,0 +1,28 @@
+        searchPlaceholder={SEARCH_PLACEHOLDER}
+        subscribeUrl={SUBSCRIBE_URL}
+        donateUrl={DONATE_URL}
+      />
```
question:
routing-ui 的 `<Header>` component 有提供 default props，我們這邊還會需要傳入這些 constants 嗎？

此外， '@/constants/index' 底下和 header 有關的 constants，是不是已經移去 routing-ui pkg 裡了？

ericsen-tsai replied:
You're right, I removed the unnecessary props and constant in [fix(frontend): remove unused menu items and constants from layout components](https://github.com/kids-reporter/kids-reporter-monorepo/pull/825/commits/8e4c5a0ec4061601a283c93a2f8591a422e54577)

nickhsine reviewed (commented):
question:

目前 membership 相關服務頁面是放在 `src/modules/membership` 底下，
而 `src/modules/membership` 底下也有 `src/modules/membership/components`。
想要借機討論一下幾個 folder 的用途。

我想請你幫助我釐清以下幾點：
1. `src/modules`, `src/service` 兩個可以怎麼區分？
2. `src/modules/membership/components` 和 `src/components` 可以怎麼區分？是共用的 component 需要拉到 `src/components` 底下嗎？
3. setter component 要算是 hook ？還是算是 component？

ericsen-tsai commented:
> question:
> 
> 目前 membership 相關服務頁面是放在 `src/modules/membership` 底下， 而 `src/modules/membership` 底下也有 `src/modules/membership/components`。 想要借機討論一下幾個 folder 的用途。
> 
> 我想請你幫助我釐清以下幾點：
> 
> 1. `src/modules`, `src/service` 兩個可以怎麼區分？
> 2. `src/modules/membership/components` 和 `src/components` 可以怎麼區分？是共用的 component 需要拉到 `src/components` 底下嗎？
> 3. setter component 要算是 hook ？還是算是 component？

1. `modules` -> This corresponds to a folder inside `/app`. It's mainly used for grouping, along with utils/components/constants/types specific to that module/page or its subpages (e.g., membership / category / article etc.). In contrast, a `service` is more like a set of highly purposeful components/functions/stores written to be reusable across various pages (e.g., call-baodaozai / auth / text-to-speech etc.).

2. Just as you said, components under `modules/{module_name}` should only be shared among other components within that same module. If it needs to be used by a different module, I think moving it to `src/components` would be more suitable.

3. I personally feel it's still a component. If it were intended to be a hook, it would need to be renamed following the "useHeaderSetter" convention (which would also be a valid approach).

ericsen-tsai committed:
fix(frontend): remove unused menu items and constants from layout components

- Removed references to additional menu items, social media links, and other constants from layout components.
- Updated Header and Footer components to render without props.

ericsen-tsai committed:
fix(frontend): fix type errors

nickhsine commented:
LGTM 讚讚讚
